src/assuan.rs

```diff
@@ -118,7 +119,9 @@ impl Connection {
                 Response::Comment(comment) => debug!("< # {}", comment),
                 Response::DataLine(data_line) => {
                     let buf = data.take();
-                    data = Some(buf.unwrap_or_else(String::new) + data_line.expose_secret());
+                    let data_line_decoded =
```

Comment:
N.B. I could `map` this back into a `SecretString`, but I figure that it's already being built up into the (unprotected) `data` string, which eventually gets secretized before being returned. Let me know what you think.

Reply:
`SecretString`'s usage here is to ensure that the secret is zeroed out when dropped. `data_line_decoded` is a `Cow<str>`, so the ideal strategy here would be:
```rust
data = ...; // use data_line_decoded
if data_line_decoded.is_owned() {
    data_line_decoded.into_owned().zeroize();
}
```

Reply:
SGTM! I'll make that change.

Reply:
Looks like `Cow::is_owned` hasn't been stabilized yet, so I went with: `if let Cow::Owned(...) = ... { ... }`.